src/shared/cloudformation/activation.ts

```diff
@@ -24,43 +19,17 @@ const TEMPLATE_FILE_GLOB_PATTERN = '**/template.{yaml,yml}'
  * @param globPattern Glob pattern for files to track. Default: TEMPLATE_FILE_GLOB_PATTERN
  */
 export async function activate(
-    extensionContext: vscode.ExtensionContext,
-    globPattern: string = TEMPLATE_FILE_GLOB_PATTERN
+    extensionContext: vscode.ExtensionContext
 ): Promise<void> {
     const registry = new DefaultCloudFormationTemplateRegistry()
-    const listener = new DefaultCloudFormationTemplateRegistryListener(registry)
-    const watcher = vscode.workspace.createFileSystemWatcher(globPattern)
-    associateFileSystemWatcherWithListener(watcher, listener, globPattern)
-    const templatePaths = await vscode.workspace.findFiles(globPattern)
-    await populateRegistry(registry, templatePaths)
-    setTemplateRegistry(registry)
+    const manager = new CloudFormationTemplateRegistryManager(registry)
+    const watcher = vscode.workspace.createFileSystemWatcher(CloudFormationTemplateRegistryManager.TEMPLATE_FILE_GLOB_PATTERN)
 
-    extensionContext.subscriptions.push(listener)
-    extensionContext.subscriptions.push(watcher)
-}
+    manager.addWatcher(watcher)
+    await manager.rebuildRegistry()
 
-/**
- * Initial registry population is an asynchronous task and cannot be initiated in the constructor
- * Call this function immediately after constructing the registry class or else you'll have an empty registry
- * Handle as many template files as possible, as quickly as possible by queuing up all promises immediately
- *
- * @param registry CloudFormationTemplateRegistry to initially populate
- * @param templatePaths Paths to enter into the registry
- */
-export async function populateRegistry(registry: CloudFormationTemplateRegistry, templatePaths: vscode.Uri[]): Promise<void> {
-    const templateParsingPromises: Promise<void>[] = []
-    for (const templatePath of templatePaths) {
-        templateParsingPromises.push(
-            new Promise(async (resolve) => {
-                try {
-                    await registry.addTemplateToTemplateData(templatePath)
-                    resolve()
-                } catch (e) {
-                    resolve()
-                }
-            })
-        )
-    }
+    setTemplateRegistry(registry)
```

Comment:
There is a comment about making a singleton registry, which removes get/setTemplateRegistry. This method doesn't instantiate DefaultCloudFormationTemplateRegistry, and instead accesses the instance.